### PR TITLE
Added missing method

### PR DESCRIPTION
**Pull request by AbhishekGarg**
Added method `refresh` to `IGridCoreApi`

### CODE REVIEW

**Comment by AbhishekGarg** [comment]
The commit [ab12eb7](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/8947/commits/ab12eb7927be2b54ccb8e8caa037a57967118441) was added after this pull request, yet it was automatically added to this.

Can I do something in this regard? Or will merging the pull request merge both commits together?

PS: Sorry, don't know the details because I am a new user to Github.

**Comment by vvakame** [comment]
don't mind. please wait for response from authors.

you should use other branch when you want to split pull request.

File: ui-grid/ui-grid.d.ts [hunk 1]
```diff
@@ -861,7 +861,7 @@ declare namespace uiGrid {
          * By default it returns the `$$hashKey` property if it exists. If it doesn't it uses gridUtil.nextUid()
          * to generate one
          */
-        rowIdentity?(): any;
+        rowIdentity?(row: IGridRowOf<TEntity>): any;
```
Comment (joeskeen) [inline ui-grid/ui-grid.d.ts]: This hunk looks good :)

File: ui-grid/ui-grid.d.ts [hunk 2]
```diff
@@ -927,6 +927,12 @@ declare namespace uiGrid {
          */
         notifyDataChange(type: string): void;
         /**
+         * Refresh the rendered grid on screen.
+         *
+         * @param {boolean} [rowsAltered] Optional flag for refreshing when the number of rows has changed.
+         */
+        refresh(rowsAltered?: boolean): ng.IPromise<any>;
```
Comment (joeskeen) [inline ui-grid/ui-grid.d.ts]: I'm not seeing any reference to the `rowsAltered` parameter in the documentation.  This is what I see:

> `refresh()`
>
>Refresh the rendered grid on screen. The refresh method re-runs both the columnProcessors and the rowProcessors, as well as calling refreshCanvas to update all the grid sizing. In general you should prefer to use queueGridRefresh instead, which is basically a debounced version of refresh.
>
>If you only want to resize the grid, not regenerate all the rows and columns, you should consider directly calling refreshCanvas instead.

Could you provide a reference to the documentation supporting the parameter, and if the documentation is wrong, would you mind submitting a change to ui-grid's documentation repo and reference this pull request?

**Comment by AbhishekGarg** [comment]
@joeskeen Submitted a pull request to [https://github.com/angular-ui/ui-grid/](https://github.com/angular-ui/ui-grid/) PR #[5315](https://github.com/angular-ui/ui-grid/pull/5315)

File: ui-grid/ui-grid.d.ts (hunk 1 above)
Comment (AbhishekGarg) [inline ui-grid/ui-grid.d.ts]: Sorry for the confusion. Actually, this PR was for the `refresh` method, but I didn't know enough about Git and pushed this commit ([ab12eb7](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/8947/commits/ab12eb7927be2b54ccb8e8caa037a57967118441)) to the same branch.
Should I revert/drop this commit or will it be merged along with the other ([9346a92](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/8947/commits/9346a921e033dc4e30c2a81a6e2451f7811cc8c5))?

Comment (joeskeen) [inline ui-grid/ui-grid.d.ts]: I'm fine leaving it here, unless @btesser objects.

Comment (btesser-r7) [inline ui-grid/ui-grid.d.ts]: lgtm

**Comment by vvakame** [comment]
👀

**Comment by AbhishekGarg** [comment]
@joeskeen @vvakame ui-grid PR has been merged. Can we please merge and close this PR, and the issue.

**Pull request merged**